prime-router/src/main/kotlin/cli/tests/BasicTests.kt

```diff
@@ -66,172 +66,94 @@ class Ping : CoolTest() {
-                passed = passed && examineStepResponse(result, "route", sender.topic)
-            if (!passed) {
-                bad("***async end2end_up FAILED***: Route result invalid")
-            }
-            val receiverNames = routeResults.values.flatMap { it?.reports ?: emptyList() }
-                .map { "${it.receivingOrg}.${it.receivingOrgSvc}" }
-            expectedReceivers.forEach { receiver ->
-                if (!receiverNames.contains(receiver.fullName)) {
-                    bad("***async end2end_up FAILED***: expected ${receiver.fullName} to have received a report")
+     override val name = "end2end_up"
+     override val description = "e2e tests that verify reports are correctly received and uploaded"
+     override val status = TestStatus.SMOKE
+
+     override suspend fun run(environment: Environment, options: CoolTestOptions): Boolean {
+         initListOfGoodReceiversAndCountiesForUniversalPipeline()
+
+         ugly("Running $name asynchronously")
+         var passed = true
+
+         ugly("Starting $name Test: send ${fullELRSender.fullName} data to $allGoodCounties")
+         passed = passed and end2EndUP(
+             environment,
+             options,
+             fullELRSender,
+             listOf(hl7FullELRReceiver, fhirFullELRReceiver)
+         )
+
+         ugly("Starting $name Test: send ${elrElimsSender.fullName} data to $allGoodCounties")
+         passed = passed and end2EndUP(environment, options, elrElimsSender, listOf(elimsReceiver))
+
+         return passed
+     }
+
+     private suspend fun end2EndUP(
+         environment: Environment,
+         options: CoolTestOptions,
+         sender: Sender,
+         expectedReceivers: List<Receiver>,
+     ): Boolean {
+         var passed = true
+         val file = File("src/test/resources/fhirengine/smoketest/valid_hl7.hl7")
+         echo("Loaded datafile $file")
+
+         val (responseCode, json) =
+             HttpUtilities.postReportFile(
+                 environment,
+                 file,
+                 sender,
+                 true,
+                 options.key,
+                 payloadName = "$name ${status.description}",
+             )
+         if (responseCode != HttpURLConnection.HTTP_CREATED) {
+             bad("***end2end_up Test FAILED***: Response code $responseCode")
+             passed = false
+         } else {
+             good("Posting of async report succeeded with response code $responseCode")
+         }
+
+         echo(json)
+         passed = passed and examinePostResponse(json, false)
+         if (!passed) {
+             bad("***end2end_up FAILED***: Error in post response")
+         }
+
+         val reportId = getReportIdFromResponse(json)
+             ?: return bad("***end2end_up FAILED***: Did not find report id")
+
+         // Need just a little wait for the db to catch up
+         waitABit(10, environment)
+
+         db = WorkflowEngine().db
+         db.transact { txn ->
```

Comment:
I'm wondering if there is a way we can figure out another way to do this without having to connect to the db.  This would simplify running these tests against staging.